components/NavigationBar.tsx

```diff
@@ -0,0 +1,171 @@
+  const closeImgSrc = theme === 'dark' ? closeWhite : close;
+  const pathname = usePathname();
+
+  return (
+    <div className="fixed z-50 flex h-16 w-full justify-between bg-white-800 py-[2.2rem] dark:bg-black-200">
+      <Link
+        href={{
+          pathname: '/',
+        }}
+        className="ml-[85px] flex items-center"
+      >
+        <div className="flex items-center">
+          <Image
+            src={logo}
+            width={30}
+            height={30}
+            alt="logo"
+            className="md:h-[45px] md:w-[45px]"
+          />
+        </div>
+      </Link>
+      <DesktopNavbar
+        setShowMobileNavbar={setShowMobileNavbar}
+        showMobileNavbar={showMobileNavbar}
+        openImgSrc={openImgSrc}
+        pathname={pathname}
+      />
+
+      {showMobileNavbar && (
+        <MobileNavigation
+          setShowMobileNavbar={setShowMobileNavbar}
+          showMobileNavbar={showMobileNavbar}
+          closeImgSrc={closeImgSrc}
+          pathname={pathname}
+        />
+      )}
+    </div>
+  );
+};
+
+export default Navbar;
+
+type MobileNavigationProps = {
+  setShowMobileNavbar: Dispatch<SetStateAction<boolean>>;
+  showMobileNavbar: boolean;
+  closeImgSrc: string;
+  pathname: string;
+};
+
+type DeskTopNavbarProps = Omit<MobileNavigationProps, 'closeImgSrc'> & {
+  openImgSrc: string;
+};
+
+type NavigationRoutesProps = {
+  pathname: string;
+  showMobileNavbar: boolean;
+};
+
+const NavigationRoutes: React.FC<NavigationRoutesProps> = ({
+  pathname,
+  showMobileNavbar,
+}) => {
+  return (
+    <>
+      {routes.map((route, index) => (
+        <Link href={route.path} key={index}>
+          <p
+            className={`mx-3.5 ${
+              showMobileNavbar ? 'mt-6' : ''
+            }  flex h-full items-center text-[0.875rem] font-normal text-white-500 dark:text-white-800 ${
+              pathname === route.path &&
+              'font-semibold text-[#0252CD] dark:text-[#428DFF]'
+            }`}
+          >
+            {route.label}
+          </p>
+        </Link>
+      ))}
+    </>
+  );
+};
+
+const DesktopNavbar: React.FC<DeskTopNavbarProps> = ({
+  setShowMobileNavbar,
+  showMobileNavbar,
+  openImgSrc,
+  pathname,
+}) => {
+  return (
+    <div className="flex items-center justify-around">
+      <div className="hidden items-center md:flex">
+        <NavigationRoutes
+          pathname={pathname}
+          showMobileNavbar={showMobileNavbar}
+        />
+      </div>
+      <div className="hidden md:flex">
+        <DownloadResume />
+        <div className="mx-[2.2rem] hidden h-[1.5rem] w-[1px] bg-white-500 dark:bg-white-500 md:flex" />
+        <ThemeSwitcher />
+      </div>
+
+      {!showMobileNavbar && (
+        <Image
+          src={openImgSrc}
+          width={30}
+          height={30}
+          alt="logo"
+          className="mr-24 flex md:hidden"
+          onClick={() => setShowMobileNavbar(!showMobileNavbar)}
```

Comment:
You're creating 2 anonymous functions in this page to toggle the navbar. Consider just creating one function 
```js
const toggleMobile = () =>{
setShowMobileNavbar(!showMobileNavbar);
}
```

And using it for both onclicks->

```js
onClick={toggleMobile}
```

This cleans up your JSX and gives you a single point to modify if functionality needs to change.

Reply:
Resolved.

Thank you.